sycl/source/context.cpp

```diff
@@ -37,13 +37,26 @@ context::context(const vector_class<device> &DeviceList,
   if (DeviceList.empty()) {
     throw invalid_parameter_error("DeviceList is empty.");
   }
-  if (DeviceList[0].is_host())
-    impl = std::make_shared<detail::context_impl>(DeviceList[0], AsyncHandler);
-  else
-    // TODO also check that devices belongs to the same platform
-    impl = std::make_shared<detail::context_impl>(DeviceList, AsyncHandler);
+  auto number = std::find_if_not(
+      DeviceList.begin(), DeviceList.end(),
+      [&](const device &current_device) { return current_device.is_host(); });
+  if ( number == DeviceList.end() && DeviceList.back().is_host())
```

Comment:
Why do you need `DeviceList.back().is_host()` condition?

If I understand correctly, at this point it is guaranteed that `DeviceList` is not empty, so, if there is no non-host devices, and list is non-empty, that it is basically guaranteed that `DeviceList.back().is_host()` is always true.

Am I missing somthing?